wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -437,8 +466,10 @@ && checkPreconditionNotEmpty(false)) {
         }
         if (WRANGLER_FAIL_PIPELINE_FOR_ERROR.isEnabled(getContext())
             && onErrorStrategy.equalsIgnoreCase(ON_ERROR_FAIL_PIPELINE)) {
-          throw new Exception(
-              String.format("Errors in Wrangler Transformation - %s", errorMessages));
+          String errorReason = String.format("Errors in Wrangler Transformation - %s",
+              errorMessages);
+          throw WranglerErrorUtil.getProgramFailureExceptionDetailsFromChain(null, errorReason,
```

Comment:
similar comment here, can we please throw `ErrorUtils.getProgramFailureException` directly from here?